### PR TITLE
Cg/http client logger test

### PR DESCRIPTION
**Pull request by ChandlerGarthwaite**
This is a copy from the previous PR. Changed File name and corrected merge into test not master

### CODE REVIEW

File: src/logger/HttpLogger_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+//Testing HTTP Logger
+
+func TestCreateInstance(t *testing.T) {
+
+	//Creating a single instance
+	httpLogger := NewHttpLogger()
+	helper := GetTestHelper()
+	assert.NotNil(t, httpLogger)
+	assert.Equal(t, helper.mockAgent, httpLogger.Agent())
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:17]: you shouldn't need the test helper here you can just use `httpLogger.agent`, you are essentially testing the getter here.

File: src/logger/HttpLogger_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+}
+
+func TestCreateMultipleInstances(t *testing.T) {
+	//Creating multiple instances
+	url1 := "https://resurface.io"
+	url2 := "https://whatever.com"
+	helper := GetTestHelper()
+
+	logger1 := NewHttpLogger(url1)
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:31]: Go doesn't allow overloading to these constructors will have to have the names of the parameters in their name, specifically here the names will be `NewHttpLoggerUrl`

File: src/logger/HttpLogger_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	logger2 := NewHttpLogger(url2)
+	logger3 := NewHttpLogger(helper.demoURL)
+
+	//Logger 1
+	assert.NotNil(t, logger1)
+	assert.Equal(t, helper.mockAgent, logger1.AGENT())
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:37]: again just use `logger1.agent`

Comment (tblenahan) [reply src/logger/HttpLogger_test.go:37]: if you disagree just let me know

File: src/logger/HttpLogger_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	assert.Nil(t, logger2.Queue())
+	assert.Equal(t, url2, logger2.Url())
+
+	//Logger 3
+	assert.NotNil(t, logger3)
+	assert.Equal(t, helper.mockAgent, logger3.Agent())
+	assert.False(t, logger3.Enablable())
+	assert.False(t, logger3.Enabled())
+	assert.Nil(t, logger3.Queue())
+	assert.Nil(t, logger3.Url())
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:57]: this shouldn't be here?

File: src/logger/HttpLogger_test.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	assert.Equal(t, helper.demoURL, logger3.Url())
+
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:59]: you should still have the test using `UsageLoggers.disable/enable`

File: src/logger/HttpLogger_test.go [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+func TestHasValidAgent(t *testing.T) {
+	//Has Valid Agent Test
+	helper := GetTestHelper()
+	logger1 := NewHttpLogger()
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:65]: `httpLogger = NewHttpLoggger()`

File: src/logger/HttpLogger_test.go [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+	agent := helper.mockAgent
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:67]: `agent := httpLogger.agent`

File: src/logger/HttpLogger_test.go [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	assert.True(t, logger1.isEnabled())
+	assert.True(t, logger2.isEnabled())
+	assert.True(t, logger3.isEnabled())
+}
+
+func TestHasValidAgent(t *testing.T) {
+	//Has Valid Agent Test
+	httpLogger := NewHttpLogger()
+
+	agent := httpLogger.mockAgent
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:74]: should be `agent := httpLogger.agent`

File: src/logger/HttpLogger_test.go [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+	//Logger 3
+	assert.NotNil(t, logger3)
+	assert.Equal(t, HttpLogger.agent, logger3.Agent())
+	assert.True(t, logger3.Enablable())
+	assert.True(t, logger3.Enabled())
+	assert.Equal(t, helper.demoURL, logger3.Url())
+
+	//Testing Usage Logger
+	//Disable
+	UsageLoggers.disable()
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:66]: `UsageLoggers`'s enabled and disabled should be public so 
`UsageLoggers.Disable()`
and `UsageLoggers.Enable()`

File: src/logger/HttpLogger_test.go [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+	UsageLoggers.enable()
+	assert.True(t, UsageLoggers.isEnabled())
+	assert.True(t, logger1.isEnabled())
+	assert.True(t, logger2.isEnabled())
+	assert.True(t, logger3.isEnabled())
+}
+
+func TestHasValidAgent(t *testing.T) {
+	//Has Valid Agent Test
+	httpLogger := NewHttpLogger()
```
Comment (tblenahan) [inline src/logger/HttpLogger_test.go:81]: `httpLogger := NewHttpLogger(Options{})`

**Review by tblenahan — CHANGES_REQUESTED** [review]
A little more context for this PR would be nice in the actual PR description.

See my comments for the last few changes.

**Review by KyleNeubarth — APPROVED** [review]

**Review by tblenahan — APPROVED** [review]
Looks good! Thank you @ChandlerGarthwaite

**Pull request merged**